sycl/source/detail/device_filter.cpp

```diff
@@ -132,7 +298,8 @@ void device_filter_list::addFilter(device_filter &Filter) {
 // 2. Filter backend match exactly with the given 'Backend'
 bool device_filter_list::backendCompatible(backend Backend) {
   for (const device_filter &Filter : FilterList) {
-    backend FilterBackend = Filter.Backend;
+    backend FilterBackend =
+        Filter.Backend ? Filter.Backend.value() : backend::all;
```

Comment:
```suggestion
    backend FilterBackend = Filter.Backend.value_or(backend::all);
```
Same for the other cases.